app/src/main/java/com/example/pwmobilesdk/MainActivity.kt

```diff
@@ -262,10 +264,19 @@ class MainActivity : AppCompatActivity() {
                     } else {
                         // Allow your merchant to try another transaction
                     }
+                    scrollTextViewLog()
                 }
             })
     }
 
+    fun scrollTextViewLog() {
+        _scrollViewTextLog?.let {
```

Comment:
I think that the Scroll method invocation for `_scrollViewTextLog` can be simplified. :) 
maybe we can try the shortform too? -- just a one liner inside the method body: :)
```
_scrollViewTextLog?.fullScroll(View.FOCUS_DOWN);
```